File: Exercícios sobre POO/Preco 12/Preco em python/Produto.py
class Produto:
    def __init__(self, nome, preco) -> None:
        self.nome = nome
        self.preco = preco
    
    def __add__(self, other):
        preco = round(self.preco + other.preco, 2)
        return f"R${preco}"
    

File: Exercícios sobre POO/Preco 12/Preco em java/src/Produto.java
import static java.lang.Math.round;

public class Produto {
    private String nome;
    private double preco;

    public Produto(String nome, double preco){
        this.nome = nome;
        this.preco = preco;
    }

    public double add(Produto other){
        return round(this.preco + other.preco);
    }
}
